test/content/HasTokenUriTests.js

```diff
@@ -97,9 +98,27 @@ describe('HasTokenUri Contract Tests', () => {
                 .to.equal("arweave.net/tx/hiddentoken-1");
             expect(await testContract.tokenUri(1, 1, false))
                 .to.equal("arweave.net/tx/hiddentoken-1v2");
+            
+            // private uris
+            var tokenUris = [[1, "arweave.net/tx/publictoken-1"], [1, "arweave.net/tx/publictoken-1v2"]];
+            var events = await testContract.setPublicUri(tokenUris);
+    
+            expect(events)
+                .to.emit(testContract, 'PublicUriUpdated')
+                .withArgs(ethers.constants.AddressZero, 1, 0);
+            expect(events)
+                .to.emit(testContract, 'PublicUriUpdated')
+                .withArgs(ethers.constants.AddressZero, 1, 1);
+            
+            // check the two versions
+            expect(await testContract.tokenUri(1, 0, true))
+                .to.equal("arweave.net/tx/publictoken-1");
+            expect(await testContract.tokenUri(1, 1, true))
+                .to.equal("arweave.net/tx/publictoken-1v2");
```

Comment:
Split this out into 2 tests. 1 for multiple tokens with the same id for public uri, and another for hidden uri